File: UVa/11800/11827.py
from fractions import *

N = int(input())
for X in range (N):
    nums = [int(x) for x in input().split()]
    maxgcd = 0

    for i in range (len(nums)):
        for j in range(i + 1, len(nums)):
            maxgcd = max(maxgcd, fractions.gcd(nums[i], nums[j]))
    print(maxgcd)
    

File: UVa/11800/11827.cpp
#include <bits/stdc++.h>
#include "header.h"

int main() {
	forX() {
		string s;
		getline(cin, s);
		stringstream ss(s);
		int v, ans = 0;
		vi nums;
		while (ss >> v)
			nums.pb(v);
		drep(i, j, sz(nums), i)
			ans = max(ans, gcd(nums[i], nums[j]));
		printd(ans);
	}
}
